packages/test/test-utils/src/testContainerRuntimeFactory.ts

```diff
@@ -83,12 +84,21 @@ export const createTestContainerRuntimeFactory = (
 		}
 
 		public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void> {
-			const rootContext = runtime.createDetachedRootDataStore([this.type], "default");
+			// Back-compat - old code does not return IDataStore for rootContext.attachRuntime() call!
+			// Thus need to leverage old API createDetachedRootDataStore() that is gone in latest releases.
+			const rootContext =
+				"createDetachedRootDataStore" in runtime
+					? (runtime as any).createDetachedRootDataStore([this.type], "default")
+					: runtime.createDetachedDataStore([this.type], "default");
+
 			const rootRuntime = await this.dataStoreFactory.instantiateDataStore(
 				rootContext,
 				/* existing */ false,
 			);
-			await rootContext.attachRuntime(this.dataStoreFactory, rootRuntime);
+			const dataStore = await rootContext.attachRuntime(this.dataStoreFactory, rootRuntime);
```

Comment:
If you need the back compat above, you're probably going to want back compat here as well because returning the `IDataStore` is a recent change.

Reply:
But I do have it here. await (in JS) can be applied to any time (including undefined) - it's a noop.
The code below will use ?. to ensure that it does not crash if it got undefined.
and then assert takes undefined as allowed result.
A bit hard to follow, but more compact compared to introducing IFs and parallel flows.